ui/src/datasets.tsx

```diff
@@ -163,204 +171,304 @@ export function Datasets() {
-                name="select-all-values"
-                checked={allAttributesChecked()}
-                onChange={() =>
-                  updateExcludedAttributes((selection) => {
-                    selection.clear();
-                    if (allAttributesChecked()) {
-                      conceptSetOccurrences.forEach((occurrence) => {
-                        selection.set(
-                          occurrence.id,
-                          new Set<string>(occurrence.attributes)
-                        );
-                      });
+              <GridBox sx={{ px: 1, overflowY: "auto" }}>
+                {cohorts.length === 0 ? (
+                  <Empty
+                    maxWidth="80%"
+                    title="Cohorts are groups of people with common traits"
+                    subtitle={
+                      <>
+                        <Link
+                          variant="link"
+                          underline="hover"
+                          onClick={showNewCohort}
+                          sx={{ cursor: "pointer" }}
+                        >
+                          Create a new cohort
+                        </Link>{" "}
+                        to define criteria
+                      </>
                     }
-                  })
-                }
-              />
-              <Typography variant="body2">
-                {allAttributesChecked() ? "Deselect all" : "Select all"}
-              </Typography>
-            </Stack>
-          </Stack>
-          <Paper
-            sx={{ p: 1, overflowY: "auto", display: "block", height: "300px" }}
-          >
-            {conceptSetOccurrences.length === 0 && (
-              <Empty
-                maxWidth="80%"
-                title="No inputs selected"
-                subtitle="You can view the available values by selecting at least one cohort and data feature"
-              />
-            )}
-            {conceptSetOccurrences.map((occurrence) => (
-              <Fragment key={occurrence.id}>
-                <Typography variant="body2em">{occurrence.name}</Typography>
-                {occurrence.attributes.map((attribute) => (
-                  <Stack key={attribute} direction="row" alignItems="center">
-                    <Checkbox
-                      size="small"
-                      fontSize="inherit"
-                      name={occurrence.id + "-" + attribute}
-                      checked={
-                        !excludedAttributes.get(occurrence.id)?.has(attribute)
-                      }
-                      onChange={() =>
-                        updateExcludedAttributes((selection) => {
-                          if (!selection?.get(occurrence.id)) {
-                            selection?.set(occurrence.id, new Set<string>());
+                  />
+                ) : (
+                  cohorts
+                    .filter((cohort) => cohort.underlayName === underlay.name)
```

Comment:
This is filtering the cohorts in a study to show only the ones for the current underlay, right? Maybe this is a good candidate for an optional filter on the cohort/concept set list endpoints. Similar to the metadata filter I added for studies.